File: 3379-transformed-array/3379-transformed-array.py
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        result = [0] * n
        for i in range(0, n):
            result[i] = nums[(i + nums[i]) % n]
        
        return result

# array, simulation
# time: O(n)
# space: O(n)

File: 3379-transformed-array/3379-transformed-array.go
func constructTransformedArray(nums []int) []int {
    n := len(nums)
    result := make([]int, n)
    for i := 0; i < n; i++ {
        result[i] = nums[(((i+nums[i])%n)+n)%n]
    }
    return result
}

// array, simulation
// time: O(n)
// space: O(n)